server/src/main/java/com/broadcom/lsp/cobol/core/messages/JSONResourceBundle.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2020 Broadcom.
+ * The term "Broadcom" refers to Broadcom Inc. and/or its subsidiaries.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *    Broadcom, Inc. - initial API and implementation
+ *
+ */
+package com.broadcom.lsp.cobol.core.messages;
+
+import com.google.gson.Gson;
+
+import java.io.BufferedInputStream;
+import java.io.InputStreamReader;
+import java.util.*;
+
+/** This is a custom {@link ResourceBundle} to support JSON. */
+public class JSONResourceBundle extends ResourceBundle {
+  private final Map<?, ?> props;
```

Comment:
The current state of JSONResourceBundle is quite generic now and can be part of a core lib (if we ever go that way). 
I always get into confusion in such cases, should I keep this within the scope of the current task or make it as a lib.

Reply:
Yes, class name, docs, and implementation are belong to a generic resource. It's a good idea, especially in the case when Java doesn't have JSON resource support by default. If you want to keep this, I can recommend you to move it into some general package like `com.broadcom.lsp.cobol.core.general` (I'm not very good at naming, but you got the point).
It will be great to use gson type checker on parsing. To be able to do it you must implement other messages related resource implementation.
Well, after all, we will have a general resource used in one place. As I know, we don't have plans to use JSON resources in other places. Personally, I try to avoid to create some general stuff just in case for the future. It could lead us to overengineering.

Reply:
I agree, no point in overengineering.